pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,1484 @@
+        templatize_fcn = templatize_expression
+        use_equation_environment = True
+        isSingle = True
+
+    elif isinstance(pyomo_component, (ExpressionBase, pyo.Var)):
+        objectives = []
+        constraints = []
+        expressions = [pyomo_component]
+        templatize_fcn = templatize_passthrough
+        use_equation_environment = True
+        isSingle = True
+
+    elif isinstance(pyomo_component, _BlockData):
+        objectives = [
+            obj
+            for obj in pyomo_component.component_data_objects(
+                pyo.Objective, descend_into=True, active=True
+            )
+        ]
+        constraints = [
+            con
+            for con in pyomo_component.component_objects(
+                pyo.Constraint, descend_into=True, active=True
+            )
+        ]
+        expressions = []
+        templatize_fcn = templatize_constraint
+
+    else:
+        raise ValueError(
+            "Invalid type %s passed into the latex printer"
+            % (str(type(pyomo_component)))
+        )
+
+    if isSingle:
+        temp_comp, temp_indexes = templatize_fcn(pyomo_component)
+        variableList = []
+        for v in identify_components(
+            temp_comp, [ScalarVar, _GeneralVarData, IndexedVar]
+        ):
+            if isinstance(v, _GeneralVarData):
+                v_write = v.parent_component()
+                if v_write not in ComponentSet(variableList):
+                    variableList.append(v_write)
+            else:
+                if v not in ComponentSet(variableList):
+                    variableList.append(v)
+
+        parameterList = []
+        for p in identify_components(
+            temp_comp, [ScalarParam, _ParamData, IndexedParam]
+        ):
+            if isinstance(p, _ParamData):
+                p_write = p.parent_component()
+                if p_write not in ComponentSet(parameterList):
+                    parameterList.append(p_write)
+            else:
+                if p not in ComponentSet(parameterList):
+                    parameterList.append(p)
+
+        # Will grab the sets as the expression is walked
+        setList = []
+
+    else:
+        variableList = [
+            vr
+            for vr in pyomo_component.component_objects(
+                pyo.Var, descend_into=True, active=True
+            )
+        ]
+
+        parameterList = [
+            pm
+            for pm in pyomo_component.component_objects(
+                pyo.Param, descend_into=True, active=True
+            )
+        ]
+
+        setList = [
+            st
+            for st in pyomo_component.component_objects(
+                pyo.Set, descend_into=True, active=True
+            )
+        ]
+
+    descriptorDict = {}
+    if use_short_descriptors:
+        descriptorDict['minimize'] = '\\min'
+        descriptorDict['maximize'] = '\\max'
+        descriptorDict['subject to'] = '\\text{s.t.}'
+        descriptorDict['with bounds'] = '\\text{w.b.}'
```

Comment:
I don't think we need 'w.b' or 'with bounds'. These are constraints and can just be in the same block of latex as the other constraints.

Reply:
I think it's necessary to indicate to the user what are actual pyomo constraints and what are bounds.  Plus the whole printer is now somewhat structured around this decision, so I'm going to leave it in